### ISSUE TITLE
Feature: Subscription Dashboard for Paid User

### ISSUE THREAD

**Issue by adiati98** [open-sauced/insights#1441]
### Type of feature

🍕 Feature

### Current behavior

I don't know if this is true for paid users or only for those who use coupon codes.

I upgraded access using the coupon code to access Open Sauced paid features for 12 months.
Credit card details were required in the process. But I can't find information about the subscription, such as if I already subscribed, the subscription expiration date, billing information, etc.
I believe I have access to it because I don't see the "Upgrade Access" button on the "User setting" anymore.

I also can't find information on whether the subscription is automatically extended and if paid users will get a notification when their subscription expires and choose to extend or terminate it. 

### Suggested solution

- A dashboard about the paid user's subscription/plan. This may include billing information, subscription expiry date, and maybe a payment method (credit card with ending number — perhaps the user has multiple cards and needs clarification on which card they use — and the expiry date).

- Information (maybe at FAQ?) about:
  - if users will get a notification before the subscription expires and whether they want to extend or cancel it.
  - if the subscription is automatically extended or not. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

**Comment by a0m0rajab** [comment]
Thank you for the feedback, I think this should be a high priority since it would impact the business outcome.
I am not sure if the logic for paid user and coupon users is different thought the answer for this would be with @bdougie and @brandonroberts

**Comment by brandonroberts** [comment]
@a0m0rajab yes, we have this on our list already, just needed an open issue for it. We definitely would not charge anyone after 12 months if there was no way to cancel the subscription. We'll use this issue for tracking and get an initial design up very soon

**Comment by a0m0rajab** [comment]
Thank you for answer, added need design to it, as for me, I think having a simple page right now with a form and checkbox for this would be a great prototype. I would love to work on this but would like to know is there any flags in beta to let the UI behave as a paid user?

**Comment by bdougie** [comment]
This does not need design or high priority. We can link to the stripe session from the user's settings

We need to grab context from the user's stripe-session and link to the stripe checkout page to manage their plan. The following is a quick mock up I took from the [dub.sh](https://dub.sh) settings page. 

We do not need copy exactly, but should leverage the design system to add toe the settings page. 

<img width="1023" alt="Screen Shot 2023-07-31 at 9 33 16 PM" src="https://github.com/open-sauced/insights/assets/5713670/4e36b2b9-743a-4dc0-bedc-7a81c1897683">

## API?

@brandonroberts, is the stripe session ID available for use make a link to the Stripe Checkout? For users that do not have a session, we should show the upgrade button. I don't have a lot of context but I imagine it is passing the session to the `getStripe` hook.

https://github.com/open-sauced/insights/blob/31a9a6abda62c84475266e602ad95a5b62ab5fab/components/organisms/StripeCheckoutButton/stripe-checkout-button.tsx#L8

**Comment by brandonroberts** [comment]
@bdougie yes, we can provide the Stripe Checkout information through our API.  Stripe Checkout also provides a direct link we can send them to where they can manage their subscription.

**Comment by babblebey** [comment]
Hi @brandonroberts 

In the absence of an API (does this mean `api-updates-needed` though?), with some few more details I think I can pack a feature for this, something similar to the `StripeCheckoutButton` component... That didn't look like it needed any specific api to handle the lots.

We can leverage the already in-use `stripe-js` api library, create a `StripeManageSubscription` component, within where we can export the following:

- A `ManageSubscriptionButton` component that configures/implements a redirect to the stripe `customer-portal` on-demand. (See [Stripe Customer Portal](https://stripe.com/docs/customer-management/integrate-customer-portal))
- A `subscription_details` state/value that carries information about the contributors' subscription, like billing cycles etc.. maybe literally the [stripe subscription object](https://stripe.com/docs/api/subscriptions/object?lang=node#subscription_object)

This way we easily build the section that alternates with the Upgrade Access (merged in #1398)  `div` on the condition of user session's `hasReports` on the `/user/settings` page/route.

@bdougie @a0m0rajab 👀

**Comment by babblebey** [comment]
👀 

@brandonroberts @bdougie @a0m0rajab

**Comment by BekahHW** [comment]
I think this has been addressed based on my settings. @brandonroberts can we close this?

<img width="391" alt="image" src="https://github.com/open-sauced/app/assets/34313413/f4a92836-6baa-4df6-8c9f-ec90ec42ed21">

**Comment by bdougie** [comment]
We can provide context on the coupon expiration here.

cc @brandonroberts
